src/Traits/HasPermissions.php

```diff
@@ -205,7 +205,7 @@ protected function getPermissionCacheKey($permission = null)
             $key .= $this->getPermissionCacheString($permission);
         }
 
-        return $key;
+        return md5($key);
```

Comment:
Given this is returning a hash of the entire key string, we probably don't need the `PermissionRegistrar::$cacheKey.'.'` prefix included a couple lines above, right?

Reply:
I've pushed a new commit now where I've removed use of tags (since we use a single cache key).
I'm pushing a 2nd commit now to remove the redundant code, `getPermissionCacheKey`, `getClassCacheString` and `getPermissionCacheString`.